add info about savepoints and updated checkpoints topic

harshini-rangaswamy opened the pull request:
# What changed, and why it matters

Add information about Savepoints and updated Checkpoints topic.

ftisiot reviewed (changes requested):
Could you add the savepoints to the dictionary to avoid the errors?

harshini-rangaswamy commented:
> Could you add the savepoints to the dictionary to avoid the errors?

I have added savepoints to the dictionary file in another PR. I think it would cause a conflict if I added it again here. What is the best way to handle this?

ftisiot commented:
I believe you can add it here as well and it will be merged successfully once rebased

ftisiot commented on docs/products/flink/concepts/savepoints.rst:
```diff
@@ -0,0 +1,8 @@
+Savepoints
+==========
+
+Savepoints in Aiven for Apache Flink® are snapshots of the current state of your Flink application. This allows you to gracefully shut down your Flink application and restart it later without losing any progress or state. 
```
```suggestion
Savepoints in Aiven for Apache Flink® are snapshots of the current state of your Flink application. They are created when stopping an application deployment and allow you to restart it later on without losing any progress or state. 
```

ftisiot reviewed (changes requested):
Minor change requested

ftisiot reviewed (approved):
LGTM
